files/image_config/caclmgrd/caclmgrd

```diff
@@ -495,9 +520,44 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
 
         self.run_commands(iptables_cmds)
 
+    def check_and_update_control_plane_acls(self, namespace, num_changes):
+        """
+        This function is intended to be spawned in a separate thread.
+        Its purpose is to prevent unnecessary iptables updates if we receive
+        multiple rapid ACL table update notifications. It will refrain from
+        updating iptables if any newer updates were detected within the delay window
+        and spawn a new thread to check again. This process will repeat until there
+        have been no changes during the delay window, at which point it will update
+        iptables using the current ACL rules.
+        """
+        while True:
+            # Sleep for our delay interval
+            time.sleep(self.UPDATE_DELAY_SECS)
+
+            with self.lock[namespace]:
+                if self.num_changes[namespace] > num_changes:
+                    # More ACL table changes occurred since this thread was spawned
+                    # spawn a new thread with the current number of changes
+                    new_changes = self.num_changes[namespace] - num_changes
+                    self.log_info("ACL config not stable for namespace '{}': {} changes detected in the past {} seconds. Skipping update ..."
+                             .format(namespace, new_changes, self.UPDATE_DELAY_SECS))
+                    num_changes = self.num_changes[namespace]
+                else:
+                    if num_changes == self.num_changes[namespace] and num_changes > 0:
```

Comment:
@jleveque do we need num_changes > 0 check ?

Reply:
It's basically a safeguard, in case `num_changes == 0` and  `self.num_changes[namespace] == 0`. We shouldn't get here in this situation, but it we do we will hit the else case and log an error message. It could be helpful for debugging.